rabbit-escape-ui-swing/src/rabbitescape/ui/swing/SwingGameLaunch.java

```diff
@@ -381,6 +387,7 @@ public int addToken( int tileX, int tileY, Token.Type ability )
         {
             graphics.playSound( "place_token" );
         }
+        solutionRecorder.append( new PlaceTokenAction( tileX, tileY ) );
         return now;
```

Comment:
The append in GameUI is switching token type, and the append here is placing a token. I think this is fine and correct.